04.ls/ls.rb

```diff
@@ -20,6 +53,77 @@ def prepare_files(options)
   options['r'] ? sorted_files.reverse : sorted_files
 end
 
+def prepare_file_details(files)
+  file_details = []
+  blocks = 0
+
+  files.each do |filename|
+    filestat = File.stat(filename)
+    blocks += filestat.blocks
+    file_details << build_file_detail(filename, filestat)
+  end
+  [blocks, file_details]
+end
+
+def build_file_detail(filename, filestat)
+  file_permissions = build_file_permissions(filestat)
+  {
+    permission: file_permissions,
+    link: filestat.nlink,
+    owner_name: Etc.getpwuid(filestat.uid).name,
+    group_name: Etc.getgrgid(filestat.gid).name,
+    size: filestat.size,
+    timestamp: filestat.mtime.strftime('%b %e %H:%M'),
+    name: filename
+  }
+end
+
+def build_file_permissions(filestat)
+  octal_mode = filestat.mode.to_s(8)
+
+  file_permissions = [FILE_TYPE[filestat.ftype]]
+  special_permission_bits = octal_mode.slice(-4, 1)
+
+  octal_mode.chars[-3..].each_with_index do |bit, idx|
+    permission = PERMISSIONS[bit]
+    special_flag = SPECIAL_FLAGS[special_permission_bits]
+    file_permissions << if special_flag && special_flag[:idx] == idx
+                          update_permission(special_flag, permission)
+                        else
+                          permission.values_at(:r, :w, :x).join
+                        end
+  end
+
+  file_permissions.join
+end
+
+def update_permission(special_flag, permission)
+  updated_execute =
+    permission[:x] == 'x' ? special_flag[:with_x] : special_flag[:without_x]
+
+  "#{permission[:r]}#{permission[:w]}#{updated_execute}"
+end
+
+def output_long_format(blocks, file_details)
+  puts "total #{blocks / 2}"
```

Comment:
```suggestion
def output_long_format(file_details)
  puts "total #{file_details.sum(&:blocks) / 2}"
```

blocksもハッシュに含めると引数が減っていいかも

Reply:
ありがとうございます！blocksもハッシュに含めるようにしました。
メソッド化しておらず`&:blocks`を実行できなかったので、`file_details.sum {|f| f[:blocks]`のようにしました。
（現段階ではクラス化は不要のため、上記のコードでも問題ないと判断し、このようにしました。）